core/src/main/scala/org/broadinstitute/dsde/rawls/Boot.scala

```diff
@@ -99,167 +110,279 @@ object Boot extends App with LazyLogging {
     val jsonFactory = JacksonFactory.getDefaultInstance
     val clientSecrets = GoogleClientSecrets.load(jsonFactory, new StringReader(gcsConfig.getString("secrets")))
     val clientEmail = gcsConfig.getString("serviceClientEmail")
-    val gcsDAO = new HttpGoogleServicesDAO(
-      false,
-      clientSecrets,
-      clientEmail,
-      gcsConfig.getString("subEmail"),
-      gcsConfig.getString("pathToPem"),
-      gcsConfig.getString("appsDomain"),
-      gcsConfig.getString("groupsPrefix"),
-      gcsConfig.getString("appName"),
-      gcsConfig.getInt("deletedBucketCheckSeconds"),
-      gcsConfig.getString("serviceProject"),
-      gcsConfig.getString("tokenEncryptionKey"),
-      gcsConfig.getString("tokenSecretsJson"),
-      gcsConfig.getString("billingPemEmail"),
-      gcsConfig.getString("pathToBillingPem"),
-      gcsConfig.getString("billingEmail"),
-      gcsConfig.getInt("bucketLogsMaxAge"),
-      workbenchMetricBaseName = metricsPrefix,
-      proxyNamePrefix = gcsConfig.getStringOr("proxyNamePrefix", "")
-    )
-
-    val pubSubDAO = new HttpGooglePubSubDAO(
-      clientEmail,
-      gcsConfig.getString("pathToPem"),
-      gcsConfig.getString("appName"),
-      gcsConfig.getString("serviceProject"),
-      workbenchMetricBaseName = metricsPrefix
+    val hammCromwellMetadataConfig = gcsConfig.getConfig("hamm-cromwell-metadata")
+    val serviceProject = gcsConfig.getString("serviceProject")
+    val hammCromwellMetadata = HammCromwellMetadata(
+      GcsBucketName(hammCromwellMetadataConfig.getString("bucket-name")),
+      ProjectTopicName.of(serviceProject, hammCromwellMetadataConfig.getString("topic-name "))
     )
 
-    val bigQueryDAO = new HttpGoogleBigQueryDAO(
-      gcsConfig.getString("appName"),
-      Json(gcsConfig.getString("bigQueryJson")),
-      metricsPrefix
-    )
+    initAppDependencies[IO](conf).use { appDependencies =>
```

Comment:
The important change in this file is actually adding `initAppDependencies` and using `IOApp` instead of `App` since `App` will be deprecated. Majority of the diff is due to indentation change